packages/osd-pm/src/utils/targeted_build.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { rm } from 'fs/promises';
+import { resolve } from 'path';
+import { spawn } from './child_process';
+import { Project } from './project';
+import { log } from './log';
+
+const BuildTargetPresets = {
+  web: '@osd/babel-preset/webpack_preset',
+  node: '@osd/babel-preset/node_preset',
+};
+
+export type BuildTargetTypes = keyof typeof BuildTargetPresets;
+export const BuildTargets = Object.keys(BuildTargetPresets) as BuildTargetTypes[];
+
+/**
+ * Run script in the given directory
+ */
+export async function buildTargetedPackage({
+  pkg,
+  sourceMaps,
+}: {
+  pkg: Project;
+  sourceMaps?: boolean;
+}) {
+  log.debug(`[${pkg.name}] deleting old output`);
+  await rm(pkg.targetLocation, { force: true, recursive: true });
+
+  log.debug(`[${pkg.name}] generating type definitions`);
+
+  await spawn('tsc', [...(sourceMaps ? ['--declarationMap', 'true'] : [])], {
+    cwd: pkg.path,
+  });
+
+  // Generate [A], [A and B], or [A, B, and C] labels
+  const targetsDisplayLabel = pkg.buildTargets
+    .join(', ')
+    .replace(/, ([^,]+)$/, pkg.buildTargets.length > 2 ? ', and $1' : ' and $1');
+  log.debug(`[${pkg.name}] transpiling for ${targetsDisplayLabel}`);
+
+  await Promise.all([
+    ...pkg.buildTargets.map((target) =>
+      spawn(
+        'babel',
```

Comment:
cant we move to using tsc here? we are losing out on type checking without it. Or is we want to continue using babel, can we add an additional step here to also perform typechecking?

Reply:
`tsc` happens a few lines above this.